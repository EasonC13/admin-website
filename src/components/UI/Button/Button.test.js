// Button.spec.js
import React from "react";
import jest from "jest-mock";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { Button } from ".";

describe("Button component", () => {
  it("should be defined", () => {
    expect(Button).toBeDefined();
  });

  it("shows button text", () => {
    const button = renderer.create(<Button>Test Button</Button>);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it("shows pill button type", () => {
    const button = renderer.create(<Button type="pill">Test Button</Button>);
    const tree = button.toJSON();
    expect(tree.props.type).toContain("pill");
  });

  it("should call mock function when button is clicked", () => {
    enzyme.configure({ adapter: new Adapter() });
    const mockFn = jest.fn();
    const tree = enzyme.shallow(<Button onClick={mockFn}>Test Button</Button>);
    tree.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });
});
