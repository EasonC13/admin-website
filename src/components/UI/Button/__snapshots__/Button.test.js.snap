// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component shows button text 1`] = `
<button
  className="css-d1zheb-Button"
>
  Test Button
</button>
`;
